Add makefiles & github workflow

sergeytimoshin opened the pull request.

sergeytimoshin committed:
Add makefiles & github workflow

sergeytimoshin committed:
Install solana deps before CI build

sergeytimoshin committed:
fix CI path

sergeytimoshin committed:
setup yaml: remove node action

sergeytimoshin committed:
Clean up & update dependencies

sergeytimoshin committed:
Update Makefile

sergeytimoshin committed:
Update Makefile

sergeytimoshin committed:
Add swap example to Makefile

sergeytimoshin committed:
Cleaning up

sergeytimoshin committed:
Add "cargo install cargo-expand" to CI workspace setup

sergeytimoshin committed:
Build projects in the CI separately

sergeytimoshin committed:
separate ci workflows for each example

sergeytimoshin committed:
update workflow names

sergeytimoshin committed:
Add private-compressed-account CI workflow

ananas-block commented on swap/programs/swaps/src/verifying_key_swaps.rs:
```diff
@@ -31,13 +31,13 @@ pub const VERIFYINGKEY_SWAPS: Groth16Verifyingkey = Groth16Verifyingkey {
-        136, 167, 226, 95, 14, 6, 148, 5, 12, 180, 145, 0, 95, 153, 33, 36, 246, 93, 31, 34, 35,
-        51, 163, 5, 57, 190, 224, 189, 34, 186, 150, 5, 154, 41, 174, 68, 48, 42, 73, 167, 16, 212,
-        72, 19, 120, 97, 13, 76, 235, 124, 116, 201, 60, 16, 44, 156, 12, 178, 191, 27, 125, 125,
-        183, 186, 143, 19, 207, 134, 32,
```
same here we push a verifying key but not the proving key

sergeytimoshin replied:
[Removed it + added to .gitignore](https://github.com/Lightprotocol/psp-examples/pull/8/commits/6a88a0fe963013e406c624230d45f8bccab735f7)

sergeytimoshin commented on .github/workflows/encrypted-messaging.yml:
```diff
@@ -0,0 +1,32 @@
+name: encrypted-messaging
```
We can run all tests by simply calling `make` command in the root folder, actually this was the first thing I did here: https://github.com/Lightprotocol/psp-examples/actions/workflows/makefile.yml

I separated them to different workflows because in this case we can run only one test when needed and skip all the others (each test takes few minutes, all together more than 30 mins)

ananas-block replied:
makes sense

sergeytimoshin commented on private-compressed-account/Makefile:
```diff
@@ -0,0 +1,22 @@
+	rm -rf node_modules
+
+
+.PHONY: test install build clean update
```
it's just a safe way to say Makefile that the list of PHONY targets (test install build clean update) is a commands, not files or folders

ananas-block replied:
thanks

sergeytimoshin committed:
Ignore +**/verifying_key_*.rs

sergeytimoshin commented on streaming-payments/programs/streaming-payments/src/verifying_key_streaming_payments.rs:
```diff
@@ -31,13 +31,13 @@ pub const VERIFYINGKEY_PSP_PAYMENT_STREAMING: Groth16Verifyingkey = Groth16Verif
-        212, 77, 191, 157, 1, 130, 233, 160, 42, 121, 24, 155, 31, 166, 39, 29, 196, 143, 125, 22,
-        61, 87, 43, 120, 31, 100, 113, 62, 212, 134, 102, 149, 89, 245, 173, 70, 121, 81, 133, 137,
-        68, 86, 7, 1, 103, 35, 175, 234, 42, 11, 240,
+        13, 144, 221, 165, 105, 204, 117, 159, 207, 122, 75, 183, 181, 211, 206, 90, 102, 31, 156,
```
> why is the verifyingkey updated but the proving key in build-circuits is not?

Because we already ignore build-circuits. 
[I added **/verifying_key_*.rs to .gitignore ](https://github.com/Lightprotocol/psp-examples/pull/8/commits/528a968d54be68dc3c51c916018ba13b905b0c19)

sergeytimoshin committed:
remove swap/programs/swaps/src/verifying_key_swaps.rs

ananas-block commented on private-compressed-account/package.json:
```diff
@@ -11,6 +11,7 @@
     "circomlib": "^2.0.5",
     "circomlibjs": "^0.1.7",
     "@lightprotocol/zk.js": "0.3.2-alpha.14",
+    "@lightprotocol/cli": "0.3.2-alpha.20",
```
the most current version is 21

ananas-block replied:
```suggestion
    "@lightprotocol/cli": "0.3.2-alpha.21",
```

sergeytimoshin replied:
yes, but it's broken 

$ light test-validator -b && yarn ts-node src/main.ts --exit

⠧ Performing setup tasks...

Performing setup tasks...

📥 Downloading https://github.com/Lightprotocol/light-protocol/releases/download/v0.3.2/spl_noop.so...
 ████████████████████████████████████████ 100% | ETA: 0s | 19960/19960
📥 Downloading https://github.com/Lightprotocol/light-protocol/releases/download/v0.3.2/merkle_tree_program.so...
 ████████████████████████████████████████ 100% | ETA: 0s | 739072/739072
📥 Downloading https://github.com/Lightprotocol/light-protocol/releases/download/v0.3.2/verifier_program_zero.so...
 ████████████████████████████████████████ 100% | ETA: 0s | 483824/483824
📥 Downloading https://github.com/Lightprotocol/light-protocol/releases/download/v0.3.2/verifier_program_storage.so...
 ████████████████████████████████████████ 100% | ETA: 0s | 516696/516696
📥 Downloading https://github.com/Lightprotocol/light-protocol/releases/download/v0.3.2/verifier_program_one.so...
 ████████████████████████████████████████ 100% | ETA: 0s | 533240/533240
📥 Downloading https://github.com/Lightprotocol/light-protocol/releases/download/v0.3.2/verifier_program_two.so...
 ████████████████████████████████████████ 100% | ETA: 0s | 492032/492032
Executing command killall...
No matching processes belonging to you were found
killall exited with code 1
Executing command killall...
No matching processes belonging to you were found
killall exited with code 1
error Command failed with exit code 1.

ananas-block replied:
is there another error or is it an infinite loop?

ananas-block commented on encrypted-messaging/package.json:
```diff
@@ -6,19 +6,19 @@
     "@coral-xyz/anchor": "^0.28.0",
-    "@lightprotocol/cli": "^0.1.1-alpha.20",
-    "@lightprotocol/zk.js": "^0.3.2-alpha.12",
+    "@lightprotocol/cli": "0.1.1-alpha.20",
```
```suggestion
    "@lightprotocol/cli": "0.1.1-alpha.21",
```

SwenSchaeferjohann replied:
running `make` runs with having a global cli with @lightprotocol/cli/0.1.1-alpha.21 on darwin-arm64 node-v19.4.0 :)

ananas-block commented on private-payments/package.json:
```diff
@@ -8,17 +8,17 @@
   "dependencies": {
     "@coral-xyz/anchor": "^0.28.0",
-    "@lightprotocol/cli": "^0.1.1-alpha.20",
+    "@lightprotocol/cli": "0.1.1-alpha.20",
```
```suggestion
    "@lightprotocol/cli": "0.1.1-alpha.21",
```

ananas-block commented on streaming-payments/Anchor.toml:
```diff
@@ -2,14 +2,14 @@
 [provider]
 cluster = "Localnet"
-wallet = "/home/ananas/.config/solana/id.json"
+wallet = "/Users/tsv/.config/solana/id.json"
```
would this work?
```suggestion
wallet = "$HOME/.config/solana/id.json"
```

vadorovsky replied:
I think `~/.config/solana/id.json` should work

ananas-block commented on streaming-payments/package.json:
```diff
@@ -2,24 +2,24 @@
     "dependencies": {
         "@coral-xyz/anchor": "^0.28.0",
-        "@lightprotocol/cli": "^0.1.1-alpha.20",
+        "@lightprotocol/cli": "0.1.1-alpha.20",
```
```suggestion
        "@lightprotocol/cli": "0.1.1-alpha.21",
```

sergeytimoshin committed:
Switch to cli .21, rm
rock-paper-scissors/programs/rock-paper-scissors/src/verifying_key_rock_paper_scissors.rs

sergeytimoshin committed:
set '@lightprotocol/cli': '0.1.1-alpha.20'

SwenSchaeferjohann reviewed (approved):
beyond the doublecheck wrt cli version lgtm, runs locally

sergeytimoshin committed:
bump cli & zk.js versions

ananas-block commented:
Swap might be failing because verifier two dependency in Cargo.toml is not bumped.
```
'Program log: AnchorError caused by account: token_program. Error Code: InvalidProgramId. Error Number: 3008. Error Message: Program ID was not as expected.',
    'Program log: Left:',
    'Program log: KitaXMAzb8GPZcc6NW6mE7P6gV2fY3Bp8NqZWfeUwqM',
    'Program log: Right:',
    'Program log: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
 ```

sergeytimoshin committed:
swap example: bump light deps in Cargo.toml + update program accordingly

sergeytimoshin committed:
Run specific workflows only when changes occur in the corresponding folder

sergeytimoshin committed:
add option to build psp examples manuallyon github

sergeytimoshin committed:
rollback to zk.js 0.3.2-alpha.14
